src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -1279,7 +1280,7 @@ bool ReadFromMergeTree::doNotMergePartsAcrossPartitionsFinal() const
 }
 
 Pipe ReadFromMergeTree::spreadMarkRangesAmongStreamsFinal(
-    RangesInDataParts && parts_with_ranges, size_t num_streams, const Names & origin_column_names, const Names & column_names, std::optional<ActionsDAG> & out_projection)
+    RangesInDataParts && parts_with_ranges, size_t num_streams, Names & origin_column_names, const Names & column_names, std::optional<ActionsDAG> & out_projection)
```

Comment:
To me, modifying origin_column_names looks pretty suspicious.
Are we sure that it is not dangerous?
If yes - cool.
Otherwise lets copy.

Reply:
Looks safe to me: it does not cause any effects up the stream (it is created just before passing it to this func).  
Something like this is also done [here](https://github.com/ClickHouse/ClickHouse/blob/a05117b3b5d4dbc3d513031f2b94af5cb12a3041/src/Processors/QueryPlan/ReadFromMergeTree.cpp#L1490), but in our case we can avoid copying as it is not used in other places

Reply:
Consider right reference (AKA &&), it seems reasonable to use it here.
Anyway, since the method is private and used once, it is not a big deal anyway.